dotCMS/src/main/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationResource.java

```diff
@@ -0,0 +1,320 @@
+import com.fasterxml.jackson.jaxrs.json.annotation.JSONP;
+import com.google.common.annotations.VisibleForTesting;
+import com.liferay.portal.model.User;
+import java.util.List;
+import java.util.Optional;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.glassfish.jersey.media.multipart.FormDataMultiPart;
+
+@Path("/v1/service/integration")
+public class ServiceIntegrationResource {
+
+    private final WebResource webResource;
+    private ServiceIntegrationHelper helper;
+
+    @VisibleForTesting
+    public ServiceIntegrationResource(final WebResource webResource,final ServiceIntegrationHelper helper) {
+        this.webResource = webResource;
+        this.helper = helper;
+    }
+
+    public ServiceIntegrationResource() {
+       this(new WebResource(),new ServiceIntegrationHelper());
+    }
+
+    @GET
+    @Path("/")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response listAvailableServices(@Context final HttpServletRequest request,
+                                                @Context final HttpServletResponse response
+    ) {
+        try {
+            final InitDataObject initData =
+                    new WebResource.InitBuilder(webResource)
+                            .requiredBackendUser(true)
+                            .requiredFrontendUser(false)
+                            .requestAndResponse(request, response)
+                            .rejectWhenNoUser(true)
+                            .init();
+            final User user = initData.getUser();
+            final List<ServiceIntegrationView> integrationViews = helper.getAvailableDescriptorViews(user);
+            if(!integrationViews.isEmpty()){
+                return Response.ok(new ResponseEntityView(integrationViews)).build(); // 200
+            }
+            throw new DoesNotExistException("Nope. No service integration is available to configure. Try uploading a file!");
+        } catch (Exception e) {
+            Logger.error(this.getClass(), "Exception on listing available service integration descriptors.", e);
+            return ResponseUtil.mapExceptionResponse(e);
+        }
+    }
+
+    @GET
+    @Path("/{serviceKey}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response getServiceIntegrationByKey(
+            @Context final HttpServletRequest request,
+            @Context final HttpServletResponse response,
+            @PathParam("serviceKey") final String serviceKey
```

Comment:
For all method receiving an arguments, they must be log in order to have guides for support and debug proposes